Orange/widgets/unsupervised/owdistancefile.py

```diff
@@ -158,7 +160,12 @@ def parametersFromFile(self, path):
         return {"recent_paths": stored_recent_paths_prepend(self.WIDGET, r)}
 
     def canDropFile(self, path: str) -> bool:
-        return os.path.splitext(path)[1].lower() in (".dst", ".xlsx")
+        try:
+            distances = DistMatrix.from_file(path)
```

Comment:
This may take too long for big files.

Reply:
What was the file dimension again? Can you tell me which file it was - or send it? I tried 600x1000 and takes a few seconds.

What I discovered was that, unfortunately, most time is spent in opening the Excel file. Hence, the File widget will take more or less the same time as well.

Of course, with this PR, the file is read twice and we must prevent this. I'm considering decorating functions that open excel files with a lru cache of size 1 that would remember the last read excel workbook - of course checking not only the file name but also the timestamp. In this way - and given that opening the file takes majority of time - the time spent in owdistancefile would be saved in owfile because it would reuse the workbook object.

Reply:
It was 1000x1000 .xlsx.